### PR TITLE
Migration of button-pressed and button-released to new GtkGesture API.

### PR DESCRIPTION
**Pull request by Christian-Kr**
Hello to everyone,

this pr will migrate the first button-pressed and button-release to the new GtkGesture API. Please test if everything is working as before. I do not find any problem but I also might have missed something.

I will wait with all further migration to this event context, until this pr has been reviewed. This is because I want to be sure, that the way I implemented it is your preferred way and I don't need to do things twice. :-)

Just to let all of you know: I am coming from the cpp perspective. This might break some things in code structure in the way I prefer to implement it.

Looking forward to your reply.
Thanks in advance

Greetings

### CODE REVIEW

File: src/dtgtk/thumbtable.c [hunk 1]
```diff
@@ -2226,12 +2231,17 @@ dt_thumbtable_t *dt_thumbtable_new()
...
-                   G_CALLBACK(_event_button_press), table);
+
+  table->gesture_button_primary = gtk_gesture_multi_press_new(GTK_WIDGET(table->widget));
+  gtk_gesture_single_set_button(GTK_GESTURE_SINGLE(table->gesture_button_primary),
+                                GDK_BUTTON_PRIMARY);
+  g_signal_connect(G_OBJECT(table->gesture_button_primary), "pressed",
+                   G_CALLBACK(_event_button_press_primary), table);
+  g_signal_connect(G_OBJECT(table->gesture_button_primary), "released",
+                   G_CALLBACK(_event_button_release_primary), table);
+
```
Comment (ralfbrown) [inline src/dtgtk/thumbtable.c:2242]: It seems likely that this sequence of four calls will be repeated over and over for each  button-press handler.  That suggests packaging them into a function called something like `dtgtk_button_handler_new` which would take as arguments the GtkWidget, the specification of which button, the press and release callbacks, and the user-data pointer.  With that abstraction in place, the above code would read
```
table->gesture_button_primary = dtgtk_button_handler_new(table->widget, GDK_BUTTON_PRIMARY, _event_button_press_primary, _event_button_release_primary, table);
```

**Comment by TurboGit** [comment]
@Christian-Kr : Please do not merge master into topic branches. I'd prefer a rebase when needed. TIA.

File: src/dtgtk/thumbtable.c (hunk 1 above)
Comment (Christian-Kr) [reply src/dtgtk/thumbtable.c:2242]: Such a message might be of general use. Do you (or anyone else) has a short hint where to place such a method? I don't have a general overview of the whole code structure for now.

**Comment by Christian-Kr** [comment]
> @Christian-Kr : Please do not merge master into topic branches. I'd prefer a rebase when needed. TIA.

@TurboGit : I have reset the merge commit and force-pushed the branch to remote. In future I will use rebase. I would also handle #15908 like that, if it is ok for you.

Greetings

File: src/dtgtk/thumbtable.c (hunk 1 above)
Comment (ralfbrown) [reply src/dtgtk/thumbtable.c:2242]: `src/dtgtk/button.h` seems the obvious place.

**Comment by Christian-Kr** [comment]
Well, there was some additional conflict with rebase. I think I ruined it. Cause it was just a small change, I will close this pr and start with a fresh branch again.

Sry for that.

**Pull request closed without merging**